File: schema-registry-confluent-avro-serializer-consumer/messaging/src/main/java/sample/consumer/ConsumerConfig.java
package sample.consumer;

import com.example.Sensor;
import com.wefox.comms.ms.crm.attachments.messaging.CommandCreateAttachments;
import java.util.function.Consumer;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ConsumerConfig {
	private final Log logger = LogFactory.getLog(getClass());

	@Bean
	public Consumer<Sensor> process()  {
		return input -> logger.info("Consumed --> input: " + input);
	}

	@Bean
	public Consumer<CommandCreateAttachments> attachment()  {
		return input -> logger.info("Consumed --> input: " + input);
	}

}

File: schema-registry-confluent-avro-serializer-consumer/messaging/src/main/kotlin/sample/consumer/ConsumerConfig.kt
package sample.consumer

import com.example.Sensor
import com.messaging.CommandCreateEntities
import java.util.function.Consumer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class ConsumerConfig {
	private val logger = loggerFor(javaClass)

	@Bean
	fun  process():Consumer<Sensor>  {
		return Consumer<Sensor> {
				logger.info("Consumed --> input: " + it)
			}
	}

	@Bean
	fun attachment():Consumer<CommandCreateEntities> {
		return Consumer<CommandCreateEntities> {
			logger.info("consuming event {}", it)// l get dispatched to DefaultDispatcher
			logger.info("Consumed --> input: " + it)
		};
	}


}
